src/modules/RA/Viewpoint/views/List.js

```diff
@@ -1,13 +1,77 @@
 import React from 'react';
-import { List } from 'react-admin';
+import {
+  Filter,
+  List,
+  SelectArrayInput,
+  SelectInput,
+  TextInput,
+} from 'react-admin';
 
 import CommonBulkActionButtons from '../../../../components/react-admin/CommonBulkActionButtons';
 import GridList from '../../../../components/react-admin/GridList';
+import { fetchFilterOptions } from '../../ra-modules';
+import { connectAppProvider } from '../../../../components/AppProvider';
 
+
+const ListFilters = props => {
+  const [selectInputs, setSelectInputs] = React.useState([]);
```

Comment:
Have you considered using useMemo here instead of useState ?

Reply:
I will have a look to this, thanks !

Reply:
I just had a quick overview on the useMemo hook. Seems like an optimization to me. If so, can you tell me what is not optimized in this specific usage of the useSate hook ?

Reply:
It's not "just optimization", I think it can completely replace useState here. The state is only used to save the last value, which seems to be the exact use case of "useMemo". You can read [this](https://wattenberger.com/blog/react-hooks) for more details.

Reply:
Discussed with @mabhub, i didn't see the async call. So there is an other problem : the setState can be called on an unmounted component so you need to apply the pattern described here: https://stackoverflow.com/a/56747072

Reply:
I wrote another comment with the modification.